R/mainTests.r

```diff
@@ -135,22 +135,32 @@ duplicate_rows <- function(data, meta) {
 
   present_obUnits_filters <- intersect(c(acceptable_observational_units, filters, filter_groups), names(data))
 
-  dupes <- suppressMessages(data %>%
-    filter(geographic_level != "School") %>%
-    filter(geographic_level != "Institution") %>%
-    filter(geographic_level != "Planning area") %>%
-    filter(geographic_level != "Provider") %>%
-    select(present_obUnits_filters) %>%
-    get_dupes())
+  if (nrow(data %>% distinct(geographic_level)) == 1 &&
+    data$geographic_level[1] %in% geography_matrix[13:14, 1]
+  ) {
+    dupes <- suppressMessages(data %>%
+      filter(geographic_level != geography_matrix[15, 1]) %>%
+      filter(geographic_level != geography_matrix[16, 1]) %>%
+      select(present_obUnits_filters) %>%
+      get_dupes())
+  } else {
+    dupes <- suppressMessages(data %>%
+      filter(geographic_level != geography_matrix[13, 1]) %>%
+      filter(geographic_level != geography_matrix[14, 1]) %>%
+      filter(geographic_level != geography_matrix[15, 1]) %>%
+      filter(geographic_level != geography_matrix[16, 1]) %>%
+      select(present_obUnits_filters) %>%
+      get_dupes())
+  }
 
   if (nrow(dupes) > 0) {
     output <- list(
-      "message" = paste("There are", cs_num(nrow(dupes)), "duplicate rows in the data file. <br> - Note that school, provider, institution, and planning area level rows are not included in this test."),
+      "message" = paste("There are", cs_num(nrow(dupes)), "duplicate rows in the data file. <br> - Note that,", paste0(geography_matrix[13, 1], ", ", geography_matrix[14, 1], ", ", geography_matrix[15, 1], " and ", geography_matrix[16, 1]), "level rows are not included in this test."),
```

Comment:
Could add another message for when you screen a school/provider-only level file to say it's ignoring only institution and planning area - at the moment says it ignores the provider/school level rows but flags as a fail for that provider test file with only provider-level data in it
![image](https://user-images.githubusercontent.com/75032474/127828730-14a4b81e-9c7b-4dd2-bf6e-9c92e0f03283.png)

Reply:
Agree - will fix in the next commit